src/f_wipe.c

```diff
@@ -102,114 +91,177 @@ static int wipe_exit(int width, int height, int ticks)
-        if (col_y[i] == -16)
-          col_y[i] = -15;
+        int r = (M_Random() % 3) - 1;
+        curry[i] = curry[i - 1] + r;
+        if (curry[i] > 0)
+        {
+            curry[i] = 0;
+        }
+        else if (curry[i] == -16)
+        {
+            curry[i] = -15;
+        }
     }
-  return 0;
+
+    memcpy(prevy, curry, sizeof(int) * wipe_columns);
+
+    return 0;
 }
 
 static int wipe_doMelt(int width, int height, int ticks)
 {
-  boolean done = true;
-  int x, y, xfactor, yfactor;
+    boolean done = true;
 
-  while (ticks--)
-    for (x=0;x<num_columns;x++)
-      if (col_y[x]<0)
+    if (ticks > 0)
+    {
+        while (ticks--)
         {
-          col_y[x]++;
-          done = false;
+            int *temp = prevy;
+            prevy = curry;
+            curry = temp;
+
+            for (int col = 0; col < wipe_columns; ++col)
+            {
+                if (prevy[col] < 0)
+                {
+                    curry[col] = prevy[col] + 1;
+                    done = false;
+                }
+                else if (prevy[col] < WIPE_ROWS)
+                {
+                    int dy = (prevy[col] < 16) ? prevy[col] + 1 : 8;
+                    curry[col] = MIN(prevy[col] + dy, WIPE_ROWS);
+
+                    done = false;
+                }
+                else
+                {
+                    curry[col] = WIPE_ROWS;
+                }
+            }
         }
-      else if (col_y[x] < COLUMN_MAX_Y)
+    }
+    else
+    {
+        for (int col = 0; col < wipe_columns; ++col)
         {
-          int dy = (col_y[x] < 16) ? col_y[x]+1 : 8;
-          if (col_y[x]+dy >= COLUMN_MAX_Y)
-            dy = COLUMN_MAX_Y - col_y[x];
-          col_y[x] += dy;
-          done = false;
+            done &= curry[col] >= WIPE_ROWS;
         }
+    }
 
-  xfactor = (width + num_columns - 1) / num_columns;
-  yfactor = (height + COLUMN_MAX_Y - 1) / COLUMN_MAX_Y;
+    return done;
+}
 
-  for (x = 0; x < width; x++)
-  {
-    byte *s, *d;
-    int scroff = col_y[x / xfactor] * yfactor;
-    if (scroff > height)
-      scroff = height;
-
-    d = wipe_scr + x;
-    s = wipe_scr_end + x * height;
-    for (y = 0; y < scroff; ++y)
-    {
-      *d = *s;
-      d += video.pitch;
-      s++;
-    }
-    s = wipe_scr_start + x * height;
-    for (; y < height; ++y)
+int wipe_renderMelt(int width, int height, int ticks)
+{
+    boolean done = true;
+
+    // Scale up and then down to handle arbitrary dimensions with integer math
+    int vertblocksize = height * 100 / WIPE_ROWS;
+    int horizblocksize = width * 100 / wipe_columns;
+    int currcol;
+    int currcolend;
+    int currrow;
+
+    V_UseBuffer(wipe_scr);
+    V_PutBlock(0, 0, width, height, wipe_scr_end);
+    V_RestoreBuffer();
+
+    for (int col = 0; col < wipe_columns; ++col)
     {
-      *d = *s;
-      d += video.pitch;
-      s++;
+        int current = FixedToInt(
+            LerpFixed(IntToFixed(prevy[col]), IntToFixed(curry[col])));
```

Comment:
Whenever interpolation is used, like when calling `LerpFixed()`, there's always an `uncapped` check with a different code path to handle capped framerate. Currently, when using capped fps, the wipe still updates at a higher rate. Compare it to a manually set fps limit of 35 which actually updates at the correct rate.

Reply:
Should be fixed here https://github.com/fabiangreffrath/woof/pull/1903/commits/b9ed42c0790bdf53684f2342903627511ab8af6c

Reply:
Works well now, thanks.